api/client/create.go

```diff
@@ -117,6 +144,14 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 	if serverResp.statusCode == 404 && strings.Contains(err.Error(), config.Image) {
 		fmt.Fprintf(cli.err, "Unable to find image '%s' locally\n", ref.ImageName(repo))
 
+		if isTrusted() && !ref.HasDigest() && trustedRef == nil {
```

Comment:
This code is starting to duplicate quite a bit of logic related to pulling then calling create. Anyway we can refactor this flow to make it more clear.